src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -385,169 +339,43 @@ public override int GetHashCode()
-            var result = new KeccakStructRef();
-            KeccakHash.ComputeHashBytesToSpan(input, result.Bytes);
-            return result;
-        }
-
-        [DebuggerStepThrough]
-        public static KeccakStructRef Compute(string input)
-        {
-            if (string.IsNullOrWhiteSpace(input))
-            {
-                return new KeccakStructRef(Keccak.OfAnEmptyString.Bytes);
-            }
-
-            var result = new KeccakStructRef();
-            KeccakHash.ComputeHashBytesToSpan(System.Text.Encoding.UTF8.GetBytes(input), result.Bytes);
-            return result;
-        }
-
-        public bool Equals(Keccak? other)
-        {
-            if (other is null)
-            {
-                return false;
-            }
-
-            return Extensions.Bytes.AreEqual(other.Bytes, Bytes);
-        }
-
-        public bool Equals(KeccakStructRef other) => Extensions.Bytes.AreEqual(other.Bytes, Bytes);
-
-        public override bool Equals(object? obj)
-        {
-            return obj?.GetType() == typeof(Keccak) && Equals((Keccak)obj);
-        }
-
-        public override int GetHashCode()
-        {
-            return MemoryMarshal.Read<int>(Bytes);
-        }
-
-        public static bool operator ==(KeccakStructRef a, Keccak? b)
+        public static bool operator <(Keccak? k1, Keccak? k2)
         {
-            if (b is null)
-            {
-                return false;
-            }
+            if (ReferenceEquals(k1, k2)) return false;
+            if (k1 is null) return false;
+            if (k2 is null) return true;
 
-            return Extensions.Bytes.AreEqual(a.Bytes, b.Bytes);
+            return k1._keccak < k2._keccak;
         }
 
-        public static bool operator ==(Keccak? a, KeccakStructRef b)
+        public static bool operator >=(Keccak? k1, Keccak? k2)
         {
-            if (a is null)
-            {
-                return false;
-            }
-
-            return Extensions.Bytes.AreEqual(a.Bytes, b.Bytes);
-        }
+            if (ReferenceEquals(k1, k2)) return true;
+            if (k1 is null) return true;
+            if (k2 is null) return false;
 
-        public static bool operator ==(KeccakStructRef a, KeccakStructRef b)
-        {
-            return Extensions.Bytes.AreEqual(a.Bytes, b.Bytes);
+            return k1._keccak >= k2._keccak;
         }
 
-        public static bool operator !=(KeccakStructRef a, Keccak b)
+        public static bool operator <=(Keccak? k1, Keccak? k2)
         {
-            return !(a == b);
-        }
+            if (ReferenceEquals(k1, k2)) return true;
+            if (k1 is null) return k2 is null;
+            if (k2 is null) return true;
 
-        public static bool operator !=(Keccak a, KeccakStructRef b)
-        {
-            return !(a == b);
+            return k1._keccak <= k2._keccak;
         }
 
-        public static bool operator !=(KeccakStructRef a, KeccakStructRef b)
+        public byte[] ToByteArray()
```

Comment:
Please review each usage of this method. I think a lot can be changed to Span and avoid alloc. Or avoid it in other way.